packages/dds/tree/src/test/simple-tree/benchmarkUtilities.ts

```diff
@@ -0,0 +1,150 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { SchemaFactory } from "../../simple-tree/index.js";
+import { hydrate } from "./utils.js";
+
+const schemaFactory = new SchemaFactory("test");
+
+/**
+ * A recursive tree node with a single leaf value
+ */
+export class DeepTreeNode extends schemaFactory.objectRecursive("deep-tree-node", {
+	node: schemaFactory.optionalRecursive([() => DeepTreeNode]),
+	value: schemaFactory.optional(schemaFactory.number),
+}) {}
+
+/**
+ * A wide tree node with all leaves having the same value
+ */
+export class WideTreeNode extends schemaFactory.object("wide-tree-node", {
+	values: schemaFactory.array(schemaFactory.number),
+}) {}
+
+/**
+ * Make a deep tree with a single leaf value
+ */
+function makeDeepTree(depth: number, leafValue: number): DeepTreeNode {
+	return depth === 1
+		? new DeepTreeNode({ node: undefined, value: leafValue })
+		: new DeepTreeNode({ node: makeDeepTree(depth - 1, leafValue), value: undefined });
+}
+
+/**
+ * generate a deep tree with a single leaf value
+ */
+export function generateDeepSimpleTree(depth: number, leafValue: number): DeepTreeNode {
+	if (depth < 1) {
+		throw new Error("Depth must be greater than 0");
+	}
+
+	return hydrate(DeepTreeNode, makeDeepTree(depth, leafValue));
+}
+
+/**
+ * read the deep tree and return the depth and value of the leaf node
+ */
+export function readDeepSimpleTree(tree: DeepTreeNode): {
+	depth: number;
+	value: number;
+} {
+	let currentNode = tree;
+	let depth = 1;
+
+	while (currentNode.node !== undefined) {
+		depth += 1;
+		currentNode = currentNode.node;
+	}
+
+	const value = currentNode.value ?? 0;
```

Comment:
We should expect there to be a leaf value here. Rather than giving 0 if its missing, we should error.
```suggestion
	const value = currentNode.value ?? assert.fail("missing leaf value");
```

Reply:
Updated